db/tables.py

```diff
@@ -408,4 +413,46 @@ def infer_table_column_types(
             table_name,
             column_name,
             engine,
+            conn=conn,
         )
+
+
+class CreateTempTableAs(DDLElement):
+    def __init__(self, name, selectable):
+        self.name = name
+        self.selectable = selectable
+
+
+@compiler.compiles(CreateTempTableAs)
+def compile(element, compiler, **_):
+    return "CREATE TEMPORARY TABLE %s AS (%s)" % (
+        element.name,
+        compiler.sql_compiler.process(element.selectable, literal_binds=True),
+    )
+
+
+def infer_table_column_types(schema, table_name, engine):
+    table = reflect_table(table_name, schema, engine)
+    temp_name = "temp_table"
+
+    select_table = select(table)
+    with engine.connect() as conn:
+        with conn.begin():
+            conn.execute(CreateTempTableAs(temp_name, select_table))
+        with conn.begin():
+            temp_table = reflect_table(temp_name, None, conn)
```

Comment:
It seems from this that the temp table is persisting over these connections.  In that case, there isn't really a point in making it an actual temporary table.  I.e., we could just use a standard table at that point, which would make some things slightly simpler (and allow for namespacing the created table).  It also avoids confusion for our future selves, trying to figure out why we need to drop a temp table.

Reply:
Correct me if I'm wrong, but I believe this shows the table persisting across several transactions within a single connection? However, given the issue with the connection pooling (see comment below), I agree that there isn't a point to using temp tables. For namespacing, are there schemas we have reserved for this sort of work?